src/App.tsx

```diff
@@ -8,55 +8,168 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getUsers } from './api/users';
+import { SelectedUser } from './types/SelectedUser';
+import { Post } from './types/Post';
+import { getPosts } from './api/posts';
+import { SelectedPost } from './types/SelectedPost';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+
+  const [error, setError] = useState<boolean>(false);
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isVisible, setIsVisible] = useState<boolean>(false);
+  const [isOpened, setIsOpened] = useState<boolean>(false);
+
+  const [selectedUser, setSelectedUser] = useState<SelectedUser | null>(null);
+  const [selectedPost, setSelectedPost] = useState<SelectedPost | null>(null);
+
+  const selectPost = (post: Post) => {
+    if (post.id !== selectedPost?.id || null) {
+      setSelectedPost({
+        id: post.id,
+        title: post.title,
+        body: post.body,
+      });
+      setIsOpened(true);
+    } else {
+      setSelectedPost(null);
+      setIsOpened(false);
+    }
+  };
+
+  const toggle = () => {
+    setIsVisible(prevIsVisible => !prevIsVisible);
+  };
+
+  useEffect(() => {
+    setPosts([]);
+
+    const Posts = async () => {
+      if (selectedUser) {
+        setIsLoading(true);
+        setIsOpened(false);
+        try {
+          const innerPosts = await getPosts(selectedUser?.id);
+
+          setPosts(innerPosts);
+        } catch (innerError) {
+          throw new Error();
+        } finally {
+          setIsLoading(false);
+          setSelectedPost(null);
+        }
+      }
+    };
+
+    Posts();
+  }, [selectedUser]);
+
+  const selectUser = (user: User) => {
+    setIsVisible(false);
+    setSelectedUser({
+      name: user.name,
+      id: user.id,
+    });
+  };
+
+  useEffect(() => {
+    const Users = async () => {
```

Comment:
The async function name 'Users' should be in camelCase and more descriptive, such as 'loadUsers'.